hack/create-cluster.sh

```diff
@@ -0,0 +1,182 @@
+# OpenShift cluster for testing purposes
+
+set -ueo pipefail
+
+ME=$(basename "$0")
+KERBEROS_USERNAME=${KERBEROS_USERNAME:-$(whoami)}
+CLUSTER_DIR=${CLUSTER_DIR:-${HOME}/.openshift/cluster}
+QUIET=${QUIET:-0}
+FORCE=${FORCE:-0}
+SSH_KEY=
+
+usage() {
+	cat <<-EOF
+	Creates OpenShift clusters with openshift-install using recommended settings
+
+	Usage:
+	  ${ME} [flags]
+
+	Flags:
+	      --cleanup            Destroy existing cluster if necessary and exit
+	  -d, --dir string         Assets directory (default "${CLUSTER_DIR}")
+	  -f, --force              Force deletion of any existing clusters if necessary
+	  -h, --help               Help for ${ME}
+	  -q, --quiet              Don't send notifications
+	      --ssh-key-file       SSH key file to deploy to nodes (default NONE)
+	  -u, --user               Your Kerberos username (default from env KERBEROS_USERNAME or \$(whoami))
+	EOF
+}
+
+main() {
+	CLEANUP_ONLY=0
+	while [[ $# -gt 0 ]]; do
+		key="$1"
+		case $key in
+		    --cleanup)
+		    CLEANUP_ONLY=1
+		    shift # past argument
+		    ;;
+		    -d|--dir)
+		    CLUSTER_DIR="$2"
+		    shift # past argument
+		    shift # past value
+		    ;;
+		    -f|--force)
+		    FORCE=1
+		    shift # past argument
+		    ;;
+		    -h|--help)
+		    usage && exit 0
+		    ;;
+		    -q|--quiet)
+		    QUIET=1
+		    shift # past argument
+		    ;;
+		    --ssh-key-file)
+		    SSH_KEY=$(cat "$2")
+		    shift # past argument
+		    shift # past value
+		    ;;
+		    -u|--user)
+		    KERBEROS_USERNAME="$2"
+		    shift # past argument
+		    shift # past value
+		    ;;
+		esac
+	done
+
+	if [ "${CLEANUP_ONLY}" -eq 1 ]; then
+	    destroy_cluster
+	    exit 0
+	fi
+
+	if [ -f "${CLUSTER_DIR}/terraform.tfstate" ]; then
+		if [ "${FORCE}" -eq 1 ]; then
+			destroy_cluster
+		else
+			abort "$(cat <<-EOF 
+			It appears you already have a running cluster.
+			It's possible you forgot to cleanup your previous cluster.
+			To force creation of a new cluster and destroy the existing 
+			cluster use the -f flag.
+			EOF
+			)"
+		fi
+	fi
+
+	make_config
+	create_cluster
+}
+
+make_config() {
+	cat <<-EOF > "${CLUSTER_DIR}/install-config.yaml"
+	---
+	apiVersion: v1
+	baseDomain: devcluster.openshift.com
+	compute:
+	- architecture: amd64
+	  hyperthreading: Enabled
+	  name: worker
+	  platform:
+	    aws:
+	      type: m4.2xlarge
+	  replicas: 3
+	controlPlane:
+	  architecture: amd64
+	  hyperthreading: Enabled
+	  name: master
+	  platform: {}
+	  replicas: 3
+	metadata:
+	  creationTimestamp: null
+	  name: ${KERBEROS_USERNAME}-$(uuidgen | cut -d- -f1)
+	networking:
+	  clusterNetwork:
+	  - cidr: 10.128.0.0/14
+	    hostPrefix: 23
+	  machineNetwork:
+	  - cidr: 10.0.0.0/16
+	  networkType: OpenShiftSDN
+	  serviceNetwork:
+	  - 172.30.0.0/16
+	fips: false
+	platform:
+	  aws:
+	    region: us-west-2
+	publish: External
+	pullSecret: |-
+	  $(tr -d '[:space:]' < ~/.docker/config.json )
```

Comment:
this should have openshift-install pull secret which is downloaded along with installer. should that pull secret be saved in `~/.docker/config.json ` ?

Reply:
Yes. As part of the google doc pull-secrets section describes.